src/disk_buffer_pool.cpp

```diff
@@ -364,148 +288,70 @@ namespace libtorrent
 
 		mutex::scoped_lock l(m_pool_mutex);
 
-		// 0 cache_buffer_chunk_size means 'automatic' (i.e.
-		// proportional to the total disk cache size)
-		m_cache_buffer_chunk_size = sett.get_int(settings_pack::cache_buffer_chunk_size);
-#ifndef TORRENT_DISABLE_POOL_ALLOCATOR
-		// if the chunk size is set to 1, there's no point in creating a pool
-		m_want_pool_allocator = sett.get_bool(settings_pack::use_disk_cache_pool)
-			&& (m_cache_buffer_chunk_size != 1);
-		// if there are no allocated blocks, it's OK to switch allocator
-		if (m_in_use == 0)
-			m_using_pool_allocator = m_want_pool_allocator;
-#endif
-
-#if TORRENT_HAVE_MMAP
-		// if we've already allocated an mmap, we can't change
-		// anything unless there are no allocations in use
+		// we can only change the cache size when it's unused. Since the whole
+		// cache is allocated as a single blob, to support resizing otherwise
+		// would require creating copies of every block (at least dirty blocks)
+		// and update all pointers in the block_cache.
```

Comment:
I followed steven's advice, see line 338. Perhaps I should lower the limit even more though